server.js

```
@@ -0,0 +1,116 @@
+"use strict";
+
+const http = require("http");
+const fs = require("fs");
+const elemTemplate = require("./elemTemplate.js");
+const indexTemplate = require("./indexTemplate.js");
+const querystring = require("querystring");
+const PORT = 8080;
+const server = http.createServer((req, res) => {
+  req.setEncoding("utf8");
+
+  // START OF GET
+  if (req.method === "GET") {
+    let URL = req.url;
+    if (URL === "/") {
+      URL = "/index.html";
+    }
+    let getType = URL.split(".");
+    let contentType = getType[1]; // grab "html" or "css"
+    // check the public directory
+    fs.readdir("./public/", (err, dir) => {
+      if (err) {
+        return console.log(err);
+      }
+      if (!dir.includes(URL.slice(1))) {
+        // check if public directory has the specified URL
+        URL = "/404.html";
+        contentType = "html";
+      }
+
+      // read contents of the public dir
+      fs.readFile(`./public${URL}`, (err, data) => {
+        if (err) {
+          return console.log(err);
+        }
+        // write the header info
+        res.writeHead(200, {
+          "content-length": data.toString().length,
+          "content-type": `text/${contentType}`
+        });
+        // write the actual file data
+        res.write(data.toString());
+        res.end();
+        return;
+      });
+    });
+  } // END OF GET
+
+  // START OF POST
+  if (req.method === "POST") {
+    let URL = req.url;
+    // body
+    let body = ""; // the data coming from Postman
+    req.on("data", chunk => {
+      body += chunk;
+
+      req.on("end", () => {
+        fs.writeFile(
+          `./public${URL}.html`,
+          elemTemplate(querystring.parse(body)), // read data as an object
+          err => {
+            if (err) {
+              return console.log(err);
```

This should return an error to the client using either a 400/500 error